data-products/src/main/scala/org/sunbird/analytics/archival/BaseArchivalJob.scala

```diff
@@ -122,29 +126,48 @@ trait BaseArchivalJob extends BaseReportsJob with IJob with ArchivalMetaDataStor
       .withColumn("question", to_json(col("question")))
   }
 
-  def archiveBatches(batchesToArchive: Map[String, Array[BatchPartition]], data: DataFrame, archivalRequest: ArchivalRequest, request: Request)(implicit config: JobConfig): Unit = {
-    batchesToArchive.flatMap(batches => {
+  def archiveBatches(batchesToArchive: Map[String, Array[BatchPartition]], data: DataFrame, requestConfig: Request)(implicit config: JobConfig): Unit = {
+    batchesToArchive.foreach(batches => {
       val processingBatch = new AtomicInteger(batches._2.length)
       JobLogger.log(s"Started Processing to archive the data", Some(Map("batch_id" -> batches._1, "total_part_files_to_archive" -> processingBatch)))
+
       // Loop through the week_num & year batch partition
-      val res = for (batch <- batches._2.asInstanceOf[Array[BatchPartition]]) yield {
+      batches._2.map((batch: BatchPartition) => {
         val filteredDF = data.filter(col("batch_id") === batch.batchId && col("year") === batch.period.year && col("week_of_year") === batch.period.weekOfYear).select(columnWithOrder.head, columnWithOrder.tail: _*)
-        val urls = upload(filteredDF, batch) // Upload the archived files into blob store
-      //TO-DO: archival Request
-        markRequestAsSuccess(archivalRequest, request)
-       JobLogger.log(s"Data is archived and Processing the remaining part files ", None, Level.INFO)
+        val collectionId = filteredDF.first().getAs[String]("course_id")
+        var archivalRequest = getRequest(collectionId, batch.batchId, batch.period.year, batch.period.weekOfYear)
+
+        if (archivalRequest == null) {
```

Comment:
Baseclass should get the archived status and updates in the postgress. it should not worry about the core logic Implementation.